svc-mesh/service-mesh.md

```diff
@@ -0,0 +1,251 @@
+through this document.
+
+For both cases, select the workloads as _trusted_ by default. This way,
+your cluster and your service mesh will run with `runc`, and only the
+containers you choose to annotate will run with Kata Containers.
+
+### Restrictions
+
+As documented [here](https://github.com/runconduit/conduit/issues/982),
+a kernel version between 4.14.22 and 4.14.40 will cause a deadlock when
+`getsockopt()` get called with `SO_ORIGINAL_DST` option. Unfortunately,
+both service meshes use this system call with this same option from the
+proxy container running inside the VM. This means that you cannot run
+this kernel version range as the guest kernel for Kata if you want your
+service mesh to work.
+
+As mentioned when explaining the basic functioning of those service meshes,
+`iptables` are heavily used, and they need to be properly enabled through
+the guest kernel config, otherwise the __init__ container will not be able
+to perform a proper setup of the rules.
+
+## Install and deploy your service mesh
+
+### Istio
+
+As a reference, you can follow Istio [instructions](https://istio.io/docs/setup/kubernetes/quick-start/#download-and-prepare-for-the-installation).
+
+Here is a summary of what you need to install Istio on your system
+```
+$ curl -L https://git.io/getLatestIstio | sh -
+$ cd istio-*
+$ export PATH=$PWD/bin:$PATH
+```
+
+Let's now deploy Istio in the control plane of your cluster
+```
+$ kubectl apply -f install/kubernetes/istio-demo.yaml
+```
+
+In order to verify the control plane has been properly deployed, you
+can use both of those following commands
+```
+$ kubectl get svc -n istio-system
+$ kubectl get pods -n istio-system -o wide
+```
+
+### Conduit
+
+As a reference, you can follow Conduit [instructions](https://conduit.io/getting-started/index.html).
+
+Here is a summary of what you need to install Conduit on your system
+```
+$ curl https://run.conduit.io/install | sh
+$ export PATH=$PATH:$HOME/.conduit/bin
+```
+
+Let's now deploy Conduit in the control plane of your cluster
+```
+$ conduit install | kubectl apply -f -
+```
+
+In order to verify the control plane has been properly deployed, you
+can use both of those following commands
+```
+$ kubectl get svc -n conduit
+$ kubectl get pods -n conduit -o wide
+```
+
+## Inject your services with sidecars
+
+Once the control plane is running, you need a deployment defining a
+few services relying on each other. Then you need to inject the YAML
+file with the sidecar proxy, using the tools provided by each service
+mesh.
+
+If you don't have such a deployment ready, you can refer to the samples
+provided by each project.
+
+### Istio
+
+Istio provides a [bookinfo](https://istio.io/docs/guides/bookinfo/)
+sample on which you can rely on to inject their `envoy` proxy as a
+sidecar.
+
+You need to use their tool called `istioctl kube-inject` to inject
+your YAML file. We are taking their [bookinfo](https://istio.io/docs/guides/bookinfo/)
```

Comment:
Lines 139-140 suggested rewrite: ... We use their [bookinfo](https://istio.io/docs/guides/bookinfo/) sample as example:

Reply:
done